### PR TITLE
Stubs for `up-client-vsomeip-rust` UTransport implementation

### PR DESCRIPTION
**Pull request by PLeVasseur**
Howdy y'all :wave: 

These are the stubs for a UTransport implementation covering [vsomeip](https://github.com/COVESA/vsomeip).

Implementation not included, obviously :sweat_smile: 

Shout-out to Devin as I was able to use the `up-client-mqtt5-rust` skeleton to turn this out.

### CODE REVIEW

File: src/transport.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    async fn unregister_listener(&self, topic: UUri, listener: &str) -> Result<(), UStatus> {
+        // implementation goes here
+        println!("Unregistering listener: {listener} for topic: {:?}", topic);
+
+        Ok(())
+    }
+
+    async fn receive(&self, _topic: UUri) -> Result<UMessage, UStatus> {
```
Comment (devkelley) [inline src/transport.rs:55]: Validating that vsomeip would not support a 'pull' implementation for uTransport.

**Review by devkelley — COMMENTED** [review]
One comment, everything else looks good

File: src/transport.rs (hunk 1 above)
Comment (PLeVasseur) [reply src/transport.rs:55]: Don't think so. Tagging @sophokles73 to confirm.

Comment (stevenhartley) [reply src/transport.rs:55]: @PLeVasseur and @devkelley you are correct. The method can return NOT_SUPPORTED

**Review by devkelley — APPROVED** [review]

**Pull request merged**